.github/workflows/cli.yml

```diff
@@ -47,6 +47,9 @@ jobs:
       - name: Vendor Go Modules
         run: make vendor
 
+      - name: Restore go/toolchain lines of go.mod
+        run: python3 .github/utils/patch-go.mod.py
```

Comment:
Did you try setting this env var? I found a few references to this as a workaround (https://github.com/actions/setup-go/issues/420)

```
GOTOOLCHAIN=local
```

Reply:
No, I didn't knew about that one so far. I'll do some tests.

Reply:
`GOTOOLCHAIN=local` doesn't make a difference in this case (at least with Golang 1.23.3). Explicitly setting `GOTOOLCHAIN` to `go1.22.9` results in no change, but it also downloads 1.22.9 (which takes some time) and it will no longer be correct once Dependabot updates the toolchain again...

Reply:
Got it thanks for testing. It looks like Dependabot already sets `GOTOOLCHAIN=local` https://github.com/dependabot/dependabot-core/pull/8044/files 🤔